### PR TITLE
Add more info for `--type` in run_tests.py and add `android` into all type

### PR DESCRIPTION
**Pull request by eggfly**
It's a small improvement for `--type` help, for who want to use `run_tests.py`.

* https://github.com/flutter/flutter/issues/95674

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

### CODE REVIEW

File: testing/run_tests.py [hunk 1]
```diff
@@ -565,12 +566,12 @@ def main():
...
       default=False, help='Provide the sanitizer suppressions lists to the via environment to the tests.')
   parser.add_argument('--adb-path', dest='adb_path', action='store',
-      default=None, help='Provide the path of adb used for android tests.  By default it looks on $PATH.')
+      default=None, help='Provide the path of adb used for android tests. By default it looks on $PATH.')
 
   args = parser.parse_args()
 
   if args.type == 'all':
-    types = ['engine', 'dart', 'benchmarks', 'java', 'objc', 'font-subset']
+    types = all_types
```
Comment (chinmaygarde) [inline testing/run_tests.py:574]: This isn't identical to the previous version. The list now contains `android`.

**Review by chinmaygarde — COMMENTED** [review]
The doc changes lgtm. But this changes the default behavior of the script. Not sure if this is the cause of the CI failure. But I agree, this all should include `android`.

File: testing/run_tests.py (hunk 1 above)
Comment (eggfly) [reply testing/run_tests.py:574]: > The doc changes lgtm. But this changes the default behavior of the script. Not sure if this is the cause of the CI failure. But I agree, this all should include `android`.

Yes it changes the behavior, and I think it should include `android` into all, so I add the it. 

The CI failure is in shell_unittests, seems like it's not related to android tests(android test is an apk installed into the real device). Let me try the CI checking again.

**Comment by eggfly** [comment]
> The doc changes lgtm. But this changes the default behavior of the script. Not sure if this is the cause of the CI failure. But I agree, this all should include `android`.

Yes it changes the behavior, and I think it should include android into all, so I add it.

The CI failure is in shell_unittests, seems like it's not related to android tests(android test is an apk installed into a device). Let me try the CI checking again.

**Comment by eggfly** [comment]
@chinmaygarde  The last failed CI log: 
* https://cirrus-ci.com/task/4554506837950464

Now the CI test passed! I check the CI scripts seems like they are rarely using the `all` type, 

Maybe the shell_unittest  is flaky or failed and then fixed.

```
[ RUN      ] ShellTest.DiscardLayerTreeOnResize
../../flutter/shell/common/shell_unittests.cc:2373: Failure
Expected equality of these values:
  1
  external_view_embedder->GetSubmittedFrameCount()
    Which is: 2
[FATAL:flutter/fml/memory/weak_ptr.h(253)] Check failed: (checker_.checker).IsCreationThreadCurrent(). 
[ERROR:flutter/fml/backtrace.cc(116)] Caught signal SIGABRT during program execution.
Frame 0: 0x7fc33867442a abort
Frame 1: 0x26bcec9 Unknown
Frame 2: 0x26bcea8 fml::LogMessage::~LogMessage()
Frame 3: 0x260ee4c fml::WeakPtrFactory<>::CheckThreadSafety()
Frame 4: 0x2602cc9 fml::WeakPtrFactory<>::~WeakPtrFactory()
Frame 5: 0x25ad594 flutter::Shell::~Shell()
Frame 6: 0x22db9ab std::__1::default_delete<>::operator()()
Frame 7: 0x22db92c std::__1::unique_ptr<>::reset()
Frame 8: 0x22d8889 std::__1::unique_ptr<>::~unique_ptr()
Frame 9: 0x24082cd flutter::testing::ShellTest_DiscardLayerTreeOnResize_Test::TestBody()
Frame 10: 0x42b97bb testing::internal::HandleSehExceptionsInMethodIfSupported<>()
Frame 11: 0x42a9d67 testing::internal::HandleExceptionsInMethodIfSupported<>()
Frame 12: 0x42993c3 testing::Test::Run()
Frame 13: 0x4299a84 testing::TestInfo::Run()
Frame 14: 0x429a08d testing::TestSuite::Run()
Frame 15: 0x42a3851 testing::internal::UnitTestImpl::RunAllTests()
Frame 16: 0x42bcbfb testing::internal::HandleSehExceptionsInMethodIfSupported<>()
Frame 17: 0x42ab7a7 testing::internal::HandleExceptionsInMethodIfSupported<>()
Frame 18: 0x42a344f testing::UnitTest::Run()
Frame 19: 0x40d6931 RUN_ALL_TESTS()
Frame 20: 0x40d67b2 main
Frame 21: 0x7fc3386602e1 __libc_start_main
Frame 22: 0x22c532a _start
<<<<<<<<<<<<<<<<<<<<<
Command run successfully in 8.25 seconds: /tmp/clean_engine/src/out/host_debug_unopt/exe.unstripped/fml_unittests --gtest_filter=-*TimeSensitiveTest* --gtest_repeat=2 --gtest_shuffle
Running runtime_unittests in /tmp/clean_engine/src

```

**Review by zanderso — APPROVED** [review]
lgtm

**Pull request merged**